Lib/test/test_cprofile.py

```diff
@@ -155,6 +157,22 @@ def test_sort(self):
         self.assertIn(b"option -s: invalid choice: 'demo'", err)
 
 
+class TestProfilingScript(unittest.TestCase):
```

Comment:
Just a few things on the test.

You don't need a new class, this is obviously a cmdline test which can be part of `class TestCommandLine(unittest.TestCase)` - just add a new test in it.

The test script is okay, maybe just throw away the blank line - it's a test script so make it a bit compact.

About file handling. I think a more common pattern is

```python
with tempfile.NamedTemporaryFile("w", delete_on_close=False) as f:
    f.write(content)
    f.close()
    # do your thing
```